packages/rest/src/coercion/coerce-parameter.ts

```diff
@@ -0,0 +1,140 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  SchemaObject,
+  ReferenceObject,
+  isReferenceObject,
+} from '@loopback/openapi-v3-types';
+import {Validator} from './validator';
+import * as HttpErrors from 'http-errors';
+import * as debugModule from 'debug';
+
+const debug = debugModule('loopback:rest:coercion');
+
+/**
+ * Coerce the http raw data to a JavaScript type data of a parameter
+ * according to its OpenAPI schema specification.
+ *
+ * @param data The raw data get from http request
+ * @param schema The parameter's schema defined in OpenAPI specification
+ */
+export function coerceParameter(
+  data: string,
+  schema?: SchemaObject | ReferenceObject,
+) {
+  if (!schema || isReferenceObject(schema)) {
+    debug(
+      'The parameter with schema %s is not coerced since schema' +
+        'dereferrence is not supported yet.',
+      schema,
```

Comment:
We should use https://github.com/BigstickCarpet/swagger-parser now as it supports both Swagger 2.0 and OAS 3.0.

Reply:
That's cool!

Let's leave integration with swagger-parser out of scope of this pull request though.

Reply:
@raymondfeng Good point to use the swagger-parser 👍 while I think the validation for whole OpenAPI spec, should happen before we coerce the parameter, which means if the generated OpenAPI spec is invalid, the app shouldn't be able to start at all. 
So the `default` clause here is actually infeasible.